app/models/user.rb

```diff
@@ -5,32 +5,56 @@ class User < ApplicationRecord
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
 
+
   def has_membership_application?
     membership_applications.size > 0
   end
 
+
   def has_company?
     has_membership_application? && (membership_applications.select { |app| app.company }).count > 0
   end
 
+
   def membership_application
     has_membership_application? ? membership_applications.last : nil
   end
 
+
   def company
     has_company? ? membership_application.company : nil
   end
 
+
   def admin?
     admin
   end
 
+
   def is_member?
     is_member
   end
 
+
   def is_member_or_admin?
     admin? || is_member?
   end
 
+
+  def is_in_company_numbered?(company_num)
+    is_member? && !(companies.detect { |c| c.company_number == company_num }).nil?
```

Comment:
`companies` is an Array, so can't use the `where`.  (and hence the kind of ugly .detect(....) stuff)

I can make it a wee bit clearer by refactoring the ugliness into a method and giving the method a name that makes the intention clearer:

     def is_in_company_numbered?(company_num)
        is_member? && my_companies_include?(company_num)
     end

    def my_companies_include?(company_num)
      !(companies.detect { |c| c.company_number == company_num }).nil?
    end

I can't think of a way to make the 
     !(companies.detect { |c| c.company_number == company_num }).nil?
any nicer, though.  Any suggestions?